File: php/src/chapter07/question7.10/Point.php
<?php

class Point {
    private $row;
    private $column;

    public function __construct($row, $column) {
        $this->row = $row;
        $this->column = $column;
    }

    public function getRow() {
        return $this->row;
    }

    public function getColumn() {
        return $this->column;
    }

    public function __toString() {
        return $this->row . ',' . $this->column;
    }
}

File: Java/Ch 08. Recursion and Dynamic Programming/src/main/java/Q8_02_Robot_in_a_Grid/Point.java
package Q8_02_Robot_in_a_Grid;

public class Point {
	public int row, column;
	public Point(int row, int column) {
		this.row = row;
		this.column = column;
	}
	
	@Override
	public String toString() {
		return "(" + row + ", " + column + ")";
	}
	
	@Override
	 public int hashCode() {
        return this.toString().hashCode();
    }
	
	@Override
	public boolean equals(Object o) {
		if ((o instanceof Point) && (((Point) o).row == this.row) && (((Point) o).column == this.column)) {
            return true;
        } else {
            return false;
        }
	}
}
